packages/lesswrong/server/callbacks/userCallbacks.ts

```diff
@@ -195,3 +200,84 @@ getCollectionHooks("Users").editSync.add(async function usersEditCheckEmail (mod
   }
   return modifier;
 });
+
+getCollectionHooks("Users").editAsync.add(async function subscribeToForumDigest (newUser: DbUser, oldUser: DbUser) {
+  if (
+    isAnyTest ||
+    forumTypeSetting.get() !== 'EAForum' ||
+    newUser.subscribedToDigest === oldUser.subscribedToDigest
+  ) {
+    return;
+  }
+
+  const mailchimpAPIKey = mailchimpAPIKeySetting.get();
+  const mailchimpForumDigestListId = mailchimpForumDigestListIdSetting.get();
+  if (!mailchimpAPIKey || !mailchimpForumDigestListId) {
+    return;
+  }
+  if (!newUser.email) {
+    captureException(new Error(`Forum digest subscription failed: no email for user ${newUser.displayName}`))
+    return;
+  }
+  const { lat: latitude, lng: longitude, known } = userGetLocation(newUser);
+  const status = newUser.subscribedToDigest ? 'subscribed' : 'unsubscribed'; 
+  
+  const emailHash = md5(newUser.email.toLowerCase());
+
+  void fetch(`https://us8.api.mailchimp.com/3.0/lists/${mailchimpForumDigestListId}/members/${emailHash}`, {
+    method: 'PUT',
+    body: JSON.stringify({
+      email_address: newUser.email,
+      email_type: 'html', 
+      ...(known && {location: {
+        latitude,
+        longitude,
+      }}),
+      status,
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `API_KEY ${mailchimpAPIKey}`,
+    },
+  }).catch(e => {
+    captureException(e);
+  });
+});
+
+/**
+ * This callback adds all new users to an audience in Mailchimp which will be used for a forthcoming
+ * (as of 2021-08-11) drip campaign.
+ */
+getCollectionHooks("Users").newAsync.add(async function subscribeToEAForumAudience(user: DbUser) {
```

Comment:
Hmm, if we want to get displayNames here, we need to wait until they complete their profile. I could imagine:
* adding a `onProfileCompleted` hook,
* running callbacks registered to that hook on profile completion, and
* adding this callback to that hook.

But potentially that's more work that you want to do to get the drip campaign to have their name. (I think a marketer might be horrified by that, but I think marketers usually have more than two devs.)

Reply:
I dove into this for a little while, but for a reason I don't yet understand adding `userOnProfileCompleted.runCallbacksAsync` in NewUserCompleteProfile is causing hundreds of `"could not resolve [unrelated dependency] (set platform to "node" when building for node)"` build errors.

I do want to understand why that's happening, but to my mind the value add here is small enough that for now I should prioritize other things.

Reply:
How the f are we getting node dependency errors? Seems reasonable to run away from that one.